master/internal/configpolicy/task_config_policy.go

```diff
@@ -160,27 +161,90 @@ func GetMergedConstraints(ctx context.Context, workspaceID int, workloadType str
+	configPolicies, err := GetTaskConfigPolicies(context.TODO(), scope, workloadType)
+	if err != nil {
+		return 0, false, fmt.Errorf("unable to fetch task config policies: %w", err)
+	}
+
+	// Cannot update priority if priority set in invariant config.
+	if configPolicies.InvariantConfig != nil {
+		switch workloadType {
+		case model.NTSCType:
+			var configs model.CommandConfig
+			err = json.Unmarshal([]byte(*configPolicies.InvariantConfig), &configs)
+			if err != nil {
+				return 0, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
+			}
+			if configs.Resources.Priority != nil {
+				adminPriority := *configs.Resources.Priority
+				return adminPriority, false,
+					fmt.Errorf("priority set by invariant config: %w", errPriorityImmutable)
+			}
+		case model.ExperimentType:
+			var configs expconf.ExperimentConfigV0
+			err = json.Unmarshal([]byte(*configPolicies.InvariantConfig), &configs)
+			if err != nil {
+				return 0, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
+			}
+			if configs.Resources().Priority() != nil {
+				adminPriority := *configs.Resources().Priority()
+				return adminPriority, false,
+					fmt.Errorf("priority set by invariant config: %w", errPriorityImmutable)
+			}
+		default:
+			return 0, false, fmt.Errorf("workload type %s not supported", workloadType)
+		}
+	}
+
+	// Find priority constraint, if set.
+	var constraints model.Constraints
+	if configPolicies.Constraints != nil {
+		if err = json.Unmarshal([]byte(*configPolicies.Constraints), &constraints); err != nil {
+			return 0, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
+		}
+		if constraints.PriorityLimit != nil {
+			return *constraints.PriorityLimit, true, nil
+		}
+	}
+
+	return 0, false, nil
+}
+
+// PriorityUpdateAllowed returns true if the desired priority is within the task config policy limit.
+func PriorityUpdateAllowed(wkspID int, workloadType string, priority int, smallerHigher bool) (bool, error) {
 	// Check if a priority limit has been set with a constraint policy.
 	// Global policies have highest precedence.
-	limit, found, err := GetPriorityLimit(context.TODO(), nil, workloadType)
-	if err != nil {
-		return false, fmt.Errorf("unable to fetch task config policy priority limit")
+	globalLimit, globalFound, err := findAllowedPriority(nil, workloadType)
+
+	if errors.Is(err, errPriorityImmutable) && globalLimit == priority {
+		// If task config policies have updated since the workload was originally scheduled, allow users
+		// to update the priority to the new priority set by invariant config.
+		return true, nil
 	}
-	if found {
-		return priorityWithinLimit(priority, limit, smallerHigher), nil
+	if err != nil {
+		return false, err
 	}
 
 	// TODO use COALESCE instead once postgres updates are complete.
 	// Workspace policies have second precedence.
-	limit, found, err = GetPriorityLimit(context.TODO(), &wkspID, workloadType)
```

Comment:
Do we still need this function? If Im understanding correctly, `findAllowedPriority` replicates and enhances its purpose

Reply:
I don't think so. Originally Carolina had it in the validation PR, but we didn't need it there either. Removing it.